src/CMakeLists.txt

```diff
@@ -4,19 +4,19 @@
 add_subdirectory (tools)
 
 # zconf.h is generated by configure_file() in the parent CMakeLists.txt
-extra_dist_generated (zlib-1.2.11/zconf.h)
+extra_dist_generated (zlib-1.2.12/zconf.h)
 set (src_zlib_DIST
-   src/zlib-1.2.11/crc32.h
-   src/zlib-1.2.11/deflate.h
-   src/zlib-1.2.11/gzguts.h
-   src/zlib-1.2.11/inffast.h
-   src/zlib-1.2.11/inffixed.h
-   src/zlib-1.2.11/inflate.h
-   src/zlib-1.2.11/inftrees.h
-   src/zlib-1.2.11/trees.h
-   src/zlib-1.2.11/zconf.h.in
-   src/zlib-1.2.11/zlib.h
-   src/zlib-1.2.11/zutil.h
+   src/zlib-1.2.12/crc32.h
+   src/zlib-1.2.12/deflate.h
+   src/zlib-1.2.12/gzguts.h
+   src/zlib-1.2.12/inffast.h
+   src/zlib-1.2.12/inffixed.h
+   src/zlib-1.2.12/inflate.h
+   src/zlib-1.2.12/inftrees.h
+   src/zlib-1.2.12/trees.h
+   src/zlib-1.2.12/zconf.h.in
+   src/zlib-1.2.12/zlib.h
+   src/zlib-1.2.12/zutil.h
```

Comment:
The task failure on [make-release-archive](https://spruce.mongodb.com/task/mongo_c_driver_releng_make_release_archive_patch_1b9d5c179d2eefd21b055a41d6d5be24c992883f_6270478557e85a2ee8b64cf4_22_05_02_21_05_11) looks relevant:

> [2022/05/02 21:09:58.960] Repository has the following files NOT in tarball
> [2022/05/02 21:09:58.960]  zlib-1.2.12/zconf.h

This may need an additional entry for `src/zlib-1.2.12/zconf.h`.

Reply:
Looks like it's actually supposed to be generated and not committed to the repo